packages/@uppy/tus/types/index.d.ts

```diff
@@ -1,23 +1,26 @@
 import type { PluginOptions, BasePlugin } from '@uppy/core'
 import type { UploadOptions } from 'tus-js-client'
 
-  type TusUploadOptions = Pick<UploadOptions, Exclude<keyof UploadOptions,
-    | 'fingerprint'
-    | 'metadata'
-    | 'onProgress'
-    | 'onChunkComplete'
-    | 'onSuccess'
-    | 'onError'
-    | 'uploadUrl'
-    | 'uploadSize'
-  >>
+type TusUploadOptions = Pick<UploadOptions, Exclude<keyof UploadOptions,
+  | 'fingerprint'
+  | 'metadata'
+  | 'onProgress'
+  | 'onChunkComplete'
+  | 'onShouldRetry'
+  | 'onSuccess'
+  | 'onError'
+  | 'uploadUrl'
+  | 'uploadSize'
+>>
+
+type Next = (err: Error) => boolean
```

Comment:
Shouldn't this always be present? Why would we retry if there wasn't an error in the first place?

Reply:
I wanted to emphasise that `err` could be `undefined` or `null`, but yeah we should indeed require the `err` to be passed to the next function, otherwise we can't implement rate-limiting.